rifServices/src/main/java/org/sahsu/rif/services/graphics/RIFTiles.java

```diff
@@ -0,0 +1,1268 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Base64.Encoder;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.Iterator;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.Month;
+
+import org.sahsu.rif.generic.util.RIFLogger;
+import org.sahsu.rif.generic.system.RIFServiceException;
+import org.sahsu.rif.services.datastorage.common.BaseSQLManager;
+import org.apache.commons.collections.IteratorUtils;
+
+import org.sahsu.rif.generic.datastorage.SelectQueryFormatter;
+import org.sahsu.rif.generic.datastorage.SQLQueryUtility;
+import org.sahsu.rif.generic.datastorage.ms.MSSQLSelectQueryFormatter;
+import org.sahsu.rif.services.system.RIFServiceError;
+import org.sahsu.rif.services.system.RIFServiceMessages;
+import org.sahsu.rif.services.system.RIFServiceStartupOptions;
+import org.sahsu.rif.services.datastorage.common.RifWellKnownText;
+import org.sahsu.rif.services.datastorage.common.RIFTilesCache;
+import org.sahsu.rif.generic.datastorage.RIFSQLException;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import org.geotools.feature.FeatureCollection;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.geojson.feature.FeatureJSON;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+
+import org.geotools.map.FeatureLayer;
+import org.geotools.map.Layer;
+import org.geotools.map.MapContent;
+import org.geotools.map.MapViewport;
+import org.geotools.styling.SLD;
+import org.geotools.styling.Style;
+import org.geotools.renderer.lite.StreamingRenderer;
+import org.geotools.renderer.label.LabelCacheImpl;
+import org.geotools.renderer.GTRenderer;
+
+import org.geotools.geometry.GeneralEnvelope;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridCoverageFactory;
+import org.geotools.coverage.processing.CoverageProcessor;
+import org.opengis.parameter.ParameterValueGroup;
+ 
+import javax.imageio.ImageIO;
+import javax.imageio.stream.ImageOutputStream;
+
+import java.awt.image.BufferedImage;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.awt.RenderingHints;
+import java.awt.image.RenderedImage;
+import java.awt.geom.AffineTransform;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+
+import java.io.IOException;
+
+/**
+ * Create PNG tiles for geolevels with more than 5000 areas.
+ *
+ * @author		Peter Hambly
+ * @version 	1.0
+ * @since 		4.0
+ */
+public class RIFTiles {
+
+	private static final RIFLogger rifLogger = RIFLogger.getLogger();
+	private static String lineSeparator = System.getProperty("line.separator");
+	
+	private static String NULL_TOPOJSON_TILE="{\"type\": \"FeatureCollection\",\"features\":[]}";
```

Comment:
Made final so it is like a c/c++ constant which seems clearest to me. Interestingly IentelliJ factor out NULL_TOPOJSON_TILE because it was used once, but made NULL_GEOJSON_TILE final which is what I would expect it to do